esphome/components/pn7160/automation.h

```diff
@@ -0,0 +1,142 @@
+#pragma once
+
+#include "esphome/core/automation.h"
+#include "esphome/core/component.h"
+#include "esphome/components/pn7160/pn7160.h"
+
+namespace esphome {
+namespace pn7160 {
+
+class PN7160OnEmulatedTagScanTrigger : public Trigger<> {
+ public:
+  explicit PN7160OnEmulatedTagScanTrigger(PN7160 *parent) {
+    parent->add_on_emulated_tag_scan_callback([this]() { this->trigger(); });
+  }
+};
+
+class PN7160OnFinishedWriteTrigger : public Trigger<> {
+ public:
+  explicit PN7160OnFinishedWriteTrigger(PN7160 *parent) {
+    parent->add_on_finished_write_callback([this]() { this->trigger(); });
+  }
+};
+
+template<typename... Ts> class PN7160IsWritingCondition : public Condition<Ts...>, public Parented<PN7160> {
+ public:
+  bool check(Ts... x) override { return this->parent_->is_writing(); }
+};
+
+template<typename... Ts> class EmulationOffAction : public Action<Ts...> {
+ public:
+  explicit EmulationOffAction(PN7160 *a_pn7160) : pn7160_(a_pn7160) {}
+
+  void play(Ts... x) override { this->pn7160_->set_tag_emulation_off(); }
+
+ protected:
+  PN7160 *pn7160_;
+};
```

Comment:
Similar to `Conditions` these can be `Parented<>` and drop the constructor and protected field

```suggestion
template<typename... Ts> class EmulationOffAction : public Action<Ts...>, public Parented<PN7160> {
 public:
  void play(Ts... x) override { this->pn7160_->set_tag_emulation_off(); }
};
```

Reply:
Done